docker/grouper/Dockerfile

```diff
@@ -0,0 +1,132 @@
+ENV TOMCAT_TGZ_URL $TOMCAT_BASE/tomcat-$TOMCAT_MAJOR/v$TOMCAT_VERSION/bin/$TOMCAT_TGZ
+
+# Grouper environment variables.
+ENV GROUPER_VERSION 2.2.2
+ENV GROUPER_BASE_URL http://software.internet2.edu/grouper/release/$GROUPER_VERSION
+ENV GROUPER_API_TGZ grouper.apiBinary-$GROUPER_VERSION.tar.gz
+ENV GROUPER_UI_TGZ grouper.ui-$GROUPER_VERSION.tar.gz
+ENV GROUPER_WS_TGZ grouper.ws-$GROUPER_VERSION.tar.gz
+ENV GROUPER_BASE /opt/grouper
+ENV GROUPER_LOGS /var/log/grouper
+ENV GROUPER_CONF /etc/grouper
+ENV GROUPER_TEMP /tmp/grouper
+ENV GROUPER_HOME $GROUPER_BASE/api
+ENV GROUPER_UI_HOME $GROUPER_BASE/ui
+ENV GROUPER_WS_HOME $GROUPER_BASE/ws
+ENV PATH $GROUPER_HOME/bin:$PATH
+
+RUN apk add --update curl \
+    && mkdir -p /tmp/jce \
+    && curl -jksSLH "Cookie: oraclelicense=accept-securebackup-cookie" -o /tmp/jce/jce.zip \
+           http://download.oracle.com/otn-pub/java/jce/7/UnlimitedJCEPolicyJDK7.zip \
+    && unzip -d /tmp/jce -o /tmp/jce/jce.zip \
+    && cp /tmp/jce/UnlimitedJCEPolicy/*.jar /opt/jdk/jre/lib/security/ \
+    && rm -rf /tmp/jce \
+    \
+    && apk add gnupg \
+    && curl -fSL https://www.apache.org/dist/tomcat/tomcat-7/KEYS -o tomcat-keys \
+    && curl -fSL https://www.apache.org/dist/ant/KEYS -o ant-keys \
+    && gpg --import tomcat-keys \
+    && gpg --import ant-keys \
+    && rm -f tomcat-keys ant-keys \
+    \
+    && mkdir -p /tmp/tarballs \
+    && cd /tmp/tarballs \
+    \
+    && curl -fSL "$ANT_URL" -o apache-ant.tar.gz \
+    && curl -fSL "$ANT_URL.asc" -o apache-ant.tar.gz.asc \
+    && gpg --verify apache-ant.tar.gz.asc \
+    \
+    && curl -fSL "$TOMCAT_TGZ_URL" -o tomcat.tar.gz \
+    && curl -fSL "$TOMCAT_TGZ_URL.asc" -o tomcat.tar.gz.asc \
+    && gpg --verify tomcat.tar.gz.asc \
+    \
+    && curl -fSL "$GROUPER_BASE_URL/$GROUPER_API_TGZ" -o grouper-api.tar.gz \
+    && curl -fSL "$GROUPER_BASE_URL/$GROUPER_UI_TGZ" -o grouper-ui.tar.gz \
+    && curl -fSL "$GROUPER_BASE_URL/$GROUPER_WS_TGZ" -o grouper-ws.tar.gz \
+    \
+    && cd /opt \
+    && tar -xzvf /tmp/tarballs/apache-ant.tar.gz \
+    && mv "apache-ant-$ANT_VERSION" "$ANT_HOME" \
+    \
+    && cd /opt \
+    && tar -xzvf /tmp/tarballs/tomcat.tar.gz \
+    && mv "apache-tomcat-$TOMCAT_VERSION" "$CATALINA_HOME" \
+    && cd "$CATALINA_HOME" \
+    && rm bin/*.bat \
+    \
+    && mkdir -p "$GROUPER_BASE" \
+        "$GROUPER_LOGS" \
+        "$GROUPER_CONF" \
+        "$GROUPER_TEMP" \
+    \
+    && cd "$GROUPER_BASE" \
+    && tar -xzvf /tmp/tarballs/grouper-api.tar.gz \
+    && mv "grouper.apiBinary-$GROUPER_VERSION" "$GROUPER_HOME" \
+    && mv /tmp/configs/api/grouper.properties "$GROUPER_HOME/conf/" \
+    && mv /tmp/configs/api/grouper.hibernate.properties "$GROUPER_HOME/conf/" \
+    && mv /tmp/configs/api/grouper-loader.properties "$GROUPER_HOME/conf/" \
+    && mv /tmp/configs/api/subject.properties "$GROUPER_HOME/conf" \
+    && mv /tmp/configs/api/log4j.properties "$GROUPER_HOME/conf" \
```

Comment:
These files are pretty small so it probably doesn't matter, but the way this is done now they'll appear in multiple layers and thus their size will be (I believe, in this case) doubled. It looks like you're doing it this way because it's impractical to create `$GROUPER_HOME` early and just add them there in the first place, though, which seems fine.